src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1107,71 +1108,69 @@ impl Cage {
-                    let mut inodeobj = FS_METADATA.inodetable.get_mut(&inodenum).unwrap();
-                    //incrementing the ref count so that when close is executed on the dup'd file
-                    //the original file does not get a negative ref count
-                    match *inodeobj {
-                        Inode::File(ref mut normalfile_inode_obj) => {
-                            normalfile_inode_obj.refcount += 1;
-                        },
-                        Inode::Dir(ref mut dir_inode_obj) => {
-                            dir_inode_obj.refcount += 1;
-                        },
-                        Inode::CharDev(ref mut chardev_inode_obj) => {
-                            chardev_inode_obj.refcount += 1;
-                        },
-                        Inode::Socket(_) => panic!("dup: fd and inode do not match.")
-                    }
-                }
-                Pipe(pipe_filedesc_obj) => {
-                    pipe_filedesc_obj.pipe.incr_ref(pipe_filedesc_obj.flags);
-                }  
-                Socket(socket_filedesc_obj) => {
-                    if let Some(socknum) = socket_filedesc_obj.socketobjectid {
-                        NET_METADATA.socket_object_table.get_mut(&socknum).unwrap().write().0.refcnt += 1;
-                    }
-                }
-                Stream(_normalfile_filedesc_obj) => {
-                    // no stream refs
+        match filedesc_enum {
+            File(normalfile_filedesc_obj) => {
+                let inodenum = normalfile_filedesc_obj.inode;
+                let mut inodeobj = FS_METADATA.inodetable.get_mut(&inodenum).unwrap();
+                //incrementing the ref count so that when close is executed on the dup'd file
+                //the original file does not get a negative ref count
+                match *inodeobj {
+                    Inode::File(ref mut normalfile_inode_obj) => {
+                        normalfile_inode_obj.refcount += 1;
+                    },
+                    Inode::Dir(ref mut dir_inode_obj) => {
+                        dir_inode_obj.refcount += 1;
+                    },
+                    Inode::CharDev(ref mut chardev_inode_obj) => {
+                        chardev_inode_obj.refcount += 1;
+                    },
+                    Inode::Socket(_) => panic!("dup: fd and inode do not match.")
                 }
-                _ => {return syscall_error(Errno::EACCES, "dup or dup2", "can't dup the provided file");},
             }
+            Pipe(pipe_filedesc_obj) => {
+                pipe_filedesc_obj.pipe.incr_ref(pipe_filedesc_obj.flags);
+            }
+            Socket(_socket_filedesc_obj) => {
+                //we handle the closing of these on drop
+            }
+            Stream(_normalfile_filedesc_obj) => {
+                // no stream refs
+            }
+            _ => {return syscall_error(Errno::EACCES, "dup or dup2", "can't dup the provided file");},
```

Comment:
Random idea but I wonder if there is a way to change this error so that we don't have to pass the string of the system call name to the error. Maybe we can make some sort of truth table that maps some enum to the names? Might just look cleaner.

Reply:
That sounds reasonable but should absolutely be done in the scope of a different PR if and when we do it.